server/item/bow.go

```diff
@@ -0,0 +1,96 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/server/item/potion"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"time"
+)
+
+// Bow is a ranged weapon that fires arrows.
+type Bow struct{}
+
+// MaxCount always returns 1.
+func (Bow) MaxCount() int {
+	return 1
+}
+
+// DurabilityInfo ...
+func (Bow) DurabilityInfo() DurabilityInfo {
+	return DurabilityInfo{
+		MaxDurability: 385,
+		BrokenItem:    simpleItem(Stack{}),
+	}
+}
+
+// Release ...
+func (b Bow) Release(releaser Releaser, duration time.Duration, ctx *UseContext) {
+	if !releaser.GameMode().Visible() {
+		return
+	}
+
+	ticks := duration.Milliseconds() / 50
+	if ticks < 3 {
+		return
+	}
+
+	t := float64(ticks) / 20
+	force := math.Min((t*t+t*2)/3, 1)
+	if force < 0.1 {
+		return
+	}
+
+	var tip potion.Potion
+	creative := releaser.GameMode().CreativeInventory()
+	if !creative {
+		if arrow, ok := ctx.FirstFunc(func(stack Stack) bool {
+			name, _ := stack.Item().EncodeItem()
+			return name == "minecraft:arrow"
+		}); ok {
+			arr := arrow.Item().(Arrow)
+			tip = arr.Tip
+
+			ctx.DamageItem(1)
+			ctx.Consume(arrow.Grow(-arrow.Count() + 1))
+		} else {
+			return
+		}
+	}
```

Comment:
Currently not setting arrow tips for creative players

Reply:
Changed.